code/modules/food/food/snacks.dm

```diff
@@ -4341,3 +4349,1687 @@
+	var/icon/I = flat_icon
+	color = "#FFFFFF" //Some fruits use the color var. Reset this so it doesnt tint the batter
+	I.Blend(new /icon('icons/obj/food_custom.dmi', rgb(255,255,255)),ICON_ADD)
+	I.Blend(new /icon('icons/obj/food_custom.dmi', coating.icon_raw),ICON_MULTIPLY)
+	var/image/J = image(I)
+	J.alpha = 200
+	J.blend_mode = BLEND_OVERLAY
+	J.tag = "coating"
+	overlays += J
+
+	if (user)
+		user.visible_message(span("notice", "[user] dips \the [src] into \the [coating.name]"), span("notice", "You dip \the [src] into \the [coating.name]"))
+
+	return 1
+
+
+//Called by cooking machines. This is mainly intended to set properties on the food that differ between raw/cooked
+/obj/item/weapon/reagent_containers/food/snacks/proc/cook()
+	if (coating)
+		var/list/temp = overlays.Copy()
+		for (var/i in temp)
+			if (istype(i, /image))
+				var/image/I = i
+				if (I.tag == "coating")
+					temp.Remove(I)
+					break
+
+		overlays = temp
+		//Carefully removing the old raw-batter overlay
+
+		if (!flat_icon)
+			flat_icon = getFlatIcon(src)
+		var/icon/I = flat_icon
+		color = "#FFFFFF" //Some fruits use the color var
+		I.Blend(new /icon('icons/obj/food_custom.dmi', rgb(255,255,255)),ICON_ADD)
+		I.Blend(new /icon('icons/obj/food_custom.dmi', coating.icon_cooked),ICON_MULTIPLY)
+		var/image/J = image(I)
+		J.alpha = 200
+		J.tag = "coating"
+		overlays += J
+
+
+		if (do_coating_prefix == 1)
+			name = "[coating.coated_adj] [name]"
+
+	for (var/r in reagents.reagent_list)
+		var/datum/reagent/R = r
+		if (istype(R, /datum/reagent/nutriment/coating))
+			var/datum/reagent/nutriment/coating/C = R
+			C.data["cooked"] = 1
+			C.name = C.cooked_name
+
+/obj/item/weapon/reagent_containers/food/snacks/proc/on_consume(var/mob/eater, var/mob/feeder = null)
+	if(!reagents.total_volume)
+		eater.visible_message("<span class='notice'>[eater] finishes eating \the [src].</span>","<span class='notice'>You finish eating \the [src].</span>")
+
+		if (!feeder)
+			feeder = eater
+
+		feeder.drop_from_inventory(src)	//so icons update :[ //what the fuck is this????
+
+		if(trash)
+			if(ispath(trash,/obj/item))
+				var/obj/item/TrashItem = new trash(feeder)
+				feeder.put_in_hands(TrashItem)
+			else if(istype(trash,/obj/item))
+				feeder.put_in_hands(trash)
+		qdel(src)
+	return
+////////////////////////////////////////////////////////////////////////////////
+/// FOOD END
+////////////////////////////////////////////////////////////////////////////////
+
+/mob/living
+	var/composition_reagent
+	var/composition_reagent_quantity
+
+/mob/living/simple_mob/adultslime
+	composition_reagent = "slimejelly"
+
+/mob/living/carbon/slime
+	composition_reagent = "slimejelly"
+
+/mob/living/carbon/alien/diona
+	composition_reagent = "nutriment"//Dionae are plants, so eating them doesn't give animal protein
+
+/mob/living/simple_mob/slime
+	composition_reagent = "slimejelly"
+
+/mob/living/simple_mob
+	var/kitchen_tag = "animal" //Used for cooking with animals
```

Comment:
I don't see it actually used anywhere? also applies to definition for mouse below.

Reply:
```else if(istype(W,/obj/item/weapon/reagent_containers/food/snacks/variable/mob))
		var/obj/item/weapon/reagent_containers/food/snacks/variable/mob/MF = W

		switch (MF.kitchen_tag)
			if ("rodent")
				result = new /obj/item/weapon/reagent_containers/food/snacks/mouseburger(src)
				to_chat(user, "You make a mouseburger!")```

snacks.dm line 5126

Reply:
That makes sense. However, I do have a question of how it would end up interacting with our current micro holder system, since the way  it's handled right now seems to only assume that simple_mobs are the holders that can be inserted (since only they have kitchen_tag var).

Reply:
That's something I hadn't considered, I might try some tests, but... why would you want to cook a micro? iirc that was kinda pref-break-y and forbidden.

Reply:
It's not forbidden globally as long as it's done quietly and, well, within prefs. Plus some people will (and have) done it to their own sleeves just for sake of lulz/'''science'''. So making sure that interactions like that are either functioning without throwing runtimes or are properly prevented is important to do regardless.

Reply:
This is resolved, see [This gif](https://i.imgur.com/aTBwuvQ.gifl)